tests/build/build.go

```diff
@@ -403,9 +404,21 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build-ser
 					}
 				}
 				Expect(outputImage).ToNot(BeEmpty(), "output image %s of the component could not be found", outputImage)
-				isExists, err := build.DoesTagExistsInQuay(outputImage)
-				Expect(err).ShouldNot(HaveOccurred(), "error while checking if the output image %s exists in quay", outputImage)
-				Expect(isExists).To(BeTrue(), "image tag does not exists in quay")
+
+				// Wait for image to be pushed to Quay - there can be a delay after PipelineRun completion
+				Eventually(func() bool {
+					isExists, err := build.DoesTagExistsInQuay(outputImage)
+					if err != nil {
+						GinkgoWriter.Printf("Error checking if image tag exists in Quay: %v\n", err)
+						return false
+					}
+					if !isExists {
+						GinkgoWriter.Printf("Image tag %s not yet available in Quay, retrying...\n", outputImage)
+					} else {
+						GinkgoWriter.Printf("Image tag %s successfully found in Quay\n", outputImage)
+					}
+					return isExists
+				}, time.Minute*3, time.Second*10).Should(BeTrue(), fmt.Sprintf("image tag %s does not exist in quay after timeout", outputImage))
```

Comment:
for the `image tag does not exists in quay` error, I think increasing the timeout will not help here, since we are already checking pipelinerun is successful [here](https://github.com/konflux-ci/e2e-tests/blob/29506f20f8cb639ec27d9e10d5706582f9bd1cf6/tests/build/build.go#L352) , which means the image tag is already pushed, it might be due to we are fetching wrong pipelinerun [here](https://github.com/konflux-ci/e2e-tests/blob/29506f20f8cb639ec27d9e10d5706582f9bd1cf6/tests/build/build.go#L397), including correct filter (like prHeadSha) while getting the pipelinerun can help